src/main/scala/chisel3/util/circt/Synthesis.scala

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.IntrinsicModule
+import chisel3.internal.Builder
+
+import circt.Intrinsic
+
+/** A 2-to-1 mux cell intrinsic.
+  */
+private class MuxCell2Intrinsic[T <: Data](gen: T) extends IntrinsicModule("circt_muxcell2") {
+  val sel = IO(Input(UInt(1.W)))
+  val high = IO(Input(gen))
+  val low = IO(Input(gen))
+  val out = IO(Output(gen))
+}
+
+object MuxCell2 {
+
+  /** Creates an intrinsic which is lowered to a 2-to-1 MUX cell in synthesis tools.
+    * @example {{{
+    * v := MuxCell2(sel, high, low)
+    * }}}
+    */
+
+  def apply[T <: Data](sel: UInt, high: T, low: T): T = {
+    val inst = Module(new MuxCell2Intrinsic(chisel3.chiselTypeOf(high)))
```

Comment:
This, and Mux4 below, should check the inputs to make sure they're valid (and the same type), see the 4 lines 37-40 here: https://github.com/chipsalliance/chisel/blob/b0f9fe0f24a43fc7a434167807eee8f5733a65bf/core/src/main/scala/chisel3/Mux.scala#L37

Reply:
Thanks, do we need to add special handling for `DontCare` as well?

Reply:
I don't think so, if it comes up we can add it.